src/test/ui/type-alias-enum-variants/self-in-enum-definition.rs

```diff
@@ -0,0 +1,8 @@
+#[repr(u8)]
+enum Alpha {
+    V1 = 41,
+    V2 = Self::V1 as u8 + 1, // OK -- but why?
```

Comment:
Really... why?

Reply:
oh you're gonna ~~love~~hate this

1. https://github.com/rust-lang/rust/issues/49765
2. https://github.com/rust-lang/rust/pull/50072
3. https://github.com/rust-lang/rust/pull/61723

It's a peephole hack for excactly `Enum::Variant as int`

Reply:
I thought it felt familiar... 😭.

At some point we should do what @RalfJung suggested in https://github.com/rust-lang/rust/pull/61723#issuecomment-500720096 since Ralf is right that this is so incredibly niche and imo clearly a bug...

Oh well... not much to do about it here!

Reply:
Adjusted the comment to refer to #50072... Should be good to merge now. :)